src/main/scala/is/hail/methods/Aggregators.scala

```diff
@@ -495,3 +494,29 @@ class TakeByAggregator[T](var t: Type, var f: (Any) => Any, var n: Int)(implicit
 
   def copy() = new TakeByAggregator(t, f, n)
 }
+
+class VirtualDSAggregator(xmin: Double, xmax: Double, ymin: Double, ymax: Double) extends TypedAggregator[IndexedSeq[Row]] {
+
+  var _state: Set[(Int, Int)] = Set.empty[(Int, Int)]
+
+  def result = {
+    _state.map(i => {
+      val x = i._1 * ((xmax - xmin) / 10)
+      val y = i._2 * ((ymax - ymin) / 10)
+      Row(x, y)
+    }).toIndexedSeq
+  }
+
+  def seqOp(x: Any) {
+    if (x != null) {
+      val xcoord = x.asInstanceOf[Row].getDouble(0)
+      val ycoord = x.asInstanceOf[Row].getDouble(1)
+      val i = ((xcoord / ((xmax - xmin) / 10)).toInt, (ycoord / ((ymax - ymin) / 10)).toInt)
+      _state += i
+    }
+  }
+
+  def combOp(agg2: this.type) = _state = _state ++ agg2._state
```

Comment:
Some points in the two combOps might be very close to each other, and you should only take one.  Better to do something like `agg2._state.foreach { i => agg.seqOp(i) }`.

Reply:
@cseed I think calling seqOp here could lead to incorrect coordinate values, because the seqOp would take in a tuple of ints that are the bin coordinates, not the tuple of doubles that are the original coordinates.

Reply:
You're totally right, I got myself mixed up.